src/Components/ServiceScene/Breakdowns/AddToExplorationButton.tsx

```diff
@@ -0,0 +1,125 @@
+  type: string;
+  title: string;
+  id: string;
+  logoPath: string;
+  note?: string;
+  drillDownLabel?: string;
+};
+
+export class AddToExplorationButton extends SceneObjectBase<AddToExplorationButtonState> {
+  constructor(state: Omit<AddToExplorationButtonState, 'disabledLinks'>) {
+    super({ ...state, disabledLinks: [] });
+    this.addActivationHandler(this.onActivate);
+  }
+
+  private onActivate = () => {
+    getLokiDatasource(this).then((ds) => {
+      this.setState({ ds });
+    });
+  };
+
+  public static Component = ({ model }: SceneComponentProps<AddToExplorationButton>) => {
+    const { ds, frame, labelName, fieldName, disabledLinks } = model.useState();
+
+    const data = sceneGraph.getData(model);
+    const sqr = sceneGraph.findObject(data, (o) => o instanceof SceneQueryRunner) as SceneQueryRunner;
+
+    const queries = useMemo(() => {
+      return sqr?.state.queries.map((q) => ({
+        ...q,
+        expr: sceneGraph.interpolate(sqr, q.expr),
+        legendFormat: sceneGraph.interpolate(sqr, q.legendFormat),
+      }));
+    }, [sqr]);
+
+    const datasourceUid = ds?.uid;
+    const timeRange = sceneGraph.getTimeRange(model);
+
+    useMemo(() => {
+      if (frame) {
+        const filter = getFilter(frame);
+        queries?.forEach((query: DataQuery & { legendFormat: string }) => {
+          if (filter) {
+            query.legendFormat = `{{${filter.name}}}`;
+          }
+        });
+      }
+    }, [frame, queries]);
+
+    const extensionPointId = 'grafana-lokiexplore-app/metric-exploration/v1';
+    const context = useMemo<ExtensionContext | undefined>(() => {
+      if (!timeRange || !queries || !datasourceUid) {
+        return;
+      }
+      return {
+        timeRange: { ...timeRange.state.value },
+        type: 'timeseries',
+        queries,
+        datasource: { uid: datasourceUid },
+        origin: 'Explore Logs',
+        url: window.location.href,
+        id: `${JSON.stringify(queries)}${labelName}${fieldName}`,
+        title: `${labelName}${fieldName ? ` > ${fieldName}` : ''}`,
+        logoPath: LokiLogo,
+        drillDownLabel: fieldName,
+      };
+    }, [datasourceUid, timeRange, queries, labelName, fieldName]);
+
+    const { links } = usePluginLinks({ extensionPointId, context });
+    return (
+      <>
+        {links
+          .filter((link) => link.pluginId === 'grafana-explorations-app' && link.onClick)
+          .map((link) => (
+            <IconButton
+              tooltip={link.description}
+              disabled={link.category === 'disabled' || disabledLinks.includes(link.id)}
+              aria-label="extension-link-to-open-exploration"
+              key={link.id}
+              name={link.icon ?? 'panel-add'}
+              onClick={(e) => {
+                if (link.onClick) {
+                  link.onClick(e);
+                }
+                disabledLinks.push(link.id);
+                model.setState({ disabledLinks: disabledLinks });
```

Comment:
This is mutating the current state array instead of creating a new one. This can lead to unexpected behaviors.

```suggestion
                disabledLinks.push(link.id);
                model.setState({ disabledLinks: [...disabledLinks] });
```
disabledLinks.slice() also works.